lambdas/initialise-ipv-session/src/main/java/uk/gov/di/ipv/core/initialiseipvsession/InitialiseIpvSessionHandler.java

```diff
@@ -236,6 +242,7 @@ public APIGatewayProxyResponseEvent handleRequest(
 
             return ApiGatewayResponseGenerator.proxyJsonResponse(HttpStatus.SC_OK, response);
         } catch (ParseException e) {
+            // Doesn't match all the potential causes
```

Comment:
Is this WIP?

Reply:
It's not for this ticket, but I might raise a tech debt ticket to review error handling, and maybe to try to make it more standardised at the same time (since there seem to be quite a few ways one can raise an error)?

Reply:
Can you elaborate?

Reply:
Here are a few thoughts explaining why I think a review/ some carved-out time to try to understand exceptions use could help:

1. We don't want to miss potential causes like ParseException, IllegalArgumentException and exceptions extending RuntimeException when handling exceptions
2. Should we try to ensure we don't have exceptions extending RuntimeException?
3. (Something we've discussed before) Should we review errors are raised with a message relevant to the cause just once, to avoid duplicated logs
6. In InitialiseIpvSessionHandler JarValidationException is only used for it’s description despite needing to define an ErrorObject for it, so maybe we should be looking at it. I see some uses of the errorObject, but do wonder whether there's a better way of defining it. Maybe to be closer to HttpResponseExceptionWithErrorBody?
7. HttpResponseExceptionWithErrorBody use could be clearer imo - it's caught all over the code base but we only really want it to know what code to send to the client instead of a message
8. (Remove unused exceptions e.g. NoVisitedCriFoundException)
9. (Use Lombok for custom exception classes)

Bit of a menageries of thoughts sorry!

Reply:
Cool, thanks! I think you're right there could be some tech debt around error handling, feel free to raise a tech debt ticket if you identify specific changes (let me know though).

Thoughts:
1. We probably _don't_ want to handle `RuntimeException`s at the top level - a RTE should be a genuinely unexpected error, and so failing the lambda is probably correct.
2. Code review, but as above, yes they should only be for genuinely unexpected errors.
3. Yes ideally
4. Yeah, if we're only using the description then using error object is probably not that useful
5. We should probably have a consistent pattern for this - either all our exceptions should extend this, or not...
6. Sure
7. Maybe, I don't know how much it adds though